src/puya/awst/nodes.py

```diff
@@ -1499,6 +1505,43 @@ def accept(self, visitor: ModuleStatementVisitor[T]) -> T:
         return visitor.visit_logic_signature(self)
 
 
+@attrs.frozen
+class CompiledContract(Expression):
+    contract: ContractReference
+    allocation_overrides: Mapping[CompiledReferenceField, Expression] = attrs.field(
+        factory=immutabledict, converter=immutabledict
+    )
+    prefix: str | None = None
```

Comment:
maybe a quick doc string about the behaviour when `None`, and also how it combines with `template_variables`?